src/onboarding/Onboarding.js

```diff
@@ -0,0 +1,583 @@
+
+    if (nextProps.visible && !props.visible) {
+      this.setState({ stepIndex: 0 })
+    }
+
+    if (
+      nextProps.daoCreationStatus !== props.daoCreationStatus &&
+      nextProps.daoCreationStatus === 'success'
+    ) {
+      setTimeout(() => {
+        this.nextStep()
+      }, 1000)
+    }
+  }
+
+  reset = () => {
+    this.setState({ ...initialState })
+  }
+
+  getSteps() {
+    const { template } = this.state
+
+    // Prepare the configure steps to be inserted
+    const configureSteps = Templates.has(template)
+      ? Templates.get(template).screens.map(step => ({
+          ...step,
+          group: Steps.Configure,
+        }))
+      : []
+
+    return [
+      { screen: 'start', group: Steps.Start },
+      { screen: 'template', group: Steps.Template },
+      { screen: 'domain', group: Steps.Domain },
+      ...configureSteps,
+      { screen: 'sign', group: Steps.Launch },
+      { screen: 'launch', group: Steps.Launch },
+    ]
+  }
+
+  currentStep() {
+    const { stepIndex } = this.state
+    const steps = this.getSteps()
+    return steps[stepIndex] || { group: Steps.Start }
+  }
+
+  // Get the template data without the screen-by-screen structure
+  getTemplateData() {
+    const { templateData } = this.state
+
+    // Transforms:
+    //
+    // [
+    //   [ { name: 'foo', value: 1 }, … ],
+    //   [ { name: 'bar', value: 2 }, … ],
+    //   …
+    // ]
+    //
+    // Into:
+    //
+    // { foo: 1, bar: 2, … }
+    //
+    return templateData.reduce(
+      (fields, { data }) => ({
+        ...fields,
+        ...data.reduce(
+          (screenFields, { name, value }) => ({
+            ...screenFields,
+            [name]: value,
+          }),
+          {}
+        ),
+      }),
+      {}
+    )
+  }
+
+  getInitialDataFromTemplate(template) {
+    if (!Templates.has(template)) {
+      return []
+    }
+    return Templates.get(template).screens.map(({ screen, fields }) => ({
+      screen,
+      data: Object.entries(fields).map(([name, { defaultValue }]) => ({
+        name,
+        value: defaultValue(),
+      })),
+    }))
+  }
+
+  // Return a screen object from a template
+  getTemplateScreen(template, screen) {
+    if (!Templates.has(template)) {
+      return null
+    }
+    return (
+      Templates.get(template).screens.find(
+        screenData => screenData.screen === screen
+      ) || null
+    )
+  }
+
+  // Filters a field value by calling the corresponding filter on the template
+  filterConfigurationValue(template, screen, name, value) {
+    const screenData = this.getTemplateScreen(template, screen)
+    return screenData ? screenData.fields[name].filter(value) : null
+  }
+
+  // Check if the data is valid by calling validateScreen() on the template
+  validateConfigurationScreen(template, screen) {
+    const screenData = this.getTemplateScreen(template, screen)
+    return screenData
+      ? screenData.validateScreen(this.getTemplateData())
+      : false
```

Comment:
Should this be `true` instead so that we don't accidentally fail validation if we don't find the screen? I guess we should never run into this case?

Reply:
Yes it would be more correct to return true here, and check for the existence of `screenData` in `isNextEnabled()`.